File: rosprolog/examples/test_rosprolog.cpp

#include <string>
#include <iostream>

#include <ros/ros.h>
#include <rosprolog/PrologClient.h>

using namespace std;

int main(int argc, char *argv[])
{
	ros::init(argc, argv, "test_rosprolog");
	PrologClient pl;
	
	PrologQuery bdgs = pl.query("member(A, [1, 2, 3, 4]), B = ['x', A], C = foo(bar, A, B)");

	for(PrologQuery::iterator it=bdgs.begin();
	    it != bdgs.end(); it++)
	{
		PrologBindings bdg = *it;
		cout << "Found solution: " << endl;
		cout << "A = "<< bdg["A"] << endl;
		cout << "B = " << bdg["B"] << endl;
		cout << "C = " << bdg["C"] << endl;
	}
	return 0;
}

File: rosprolog/examples/test_rosprolog.py
#!/usr/bin/env python

import roslib; roslib.load_manifest('rosprolog')

import rospy
from rosprolog_client import PrologException, Prolog

if __name__ == '__main__':
    rospy.init_node('test_rosprolog')
    prolog = Prolog()
    query = prolog.query("member(A, [1, 2, 3, 4]), B = ['x', A]")
    for solution in query.solutions():
        print 'Found solution. A = %s, B = %s' % (solution['A'], solution['B'])
    query.finish()
